plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/business/internal/migration/WorkspaceImageGMFBoundsMigrationParticipant.java

```diff
@@ -89,20 +91,9 @@ protected void postLoad(DAnalysis dAnalysis, Version loadedVersion) {
                     if (descriptor.getRepresentation() instanceof DDiagram) {
                         DDiagram dDiagram = (DDiagram) descriptor.getRepresentation();
                         DDiagramGraphicalQuery query = new DDiagramGraphicalQuery(dDiagram);
-                        Option<Diagram> gmfDiagram = query.getAssociatedGMFDiagram();
-                        if (gmfDiagram.some()) {
-                            boolean migrationOccurredInCurrentDiag = false;
-                            String representationName = StringUtil.EMPTY_STRING;
-                            for (Object child : gmfDiagram.get().getChildren()) {
-                                if (resizeWorkspaceImageGMFBounds(child, previousMigrationApplied)) {
-                                    migrationOccurred = true;
-                                    migrationOccurredInCurrentDiag = true;
-                                    representationName = dDiagram.getName();
-                                }
-                            }
-                            if (migrationOccurredInCurrentDiag) {
-                                sb.append(MessageFormat.format(Messages.WorkspaceImageGMFBoundsMigrationParticipant_GMFBoundsResized, representationName));
-                            }
+                        Option<Diagram> optionalGmfDiagram = query.getAssociatedGMFDiagram();
+                        if (optionalGmfDiagram.some()) {
+                            migrationOccurred = resizeWorkspaceImageGmfBoundsOfDiagram(previousMigrationApplied, sb, dDiagram, optionalGmfDiagram.get()) || migrationOccurred;
```

Comment:
It works with lazy evaluation but it would be clearer on two lines.

Reply:
OK. I will change this.